### PR TITLE
DOM adapters that work on the server

### PR DESCRIPTION
**Pull request by yjbanov**

### CODE REVIEW

File: modules/angular2/src/core/application.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import {Injector, bind, OpaqueToken} from 'angular2/di';
 import {Type, FIELD, isBlank, isPresent, BaseException, assertionsEnabled, print} from 'angular2/src/facade/lang';
-import {DOM, Element} from 'angular2/src/facade/dom';
+import {BrowserDomAdapter} from 'angular2/src/dom/browser_adapter';
```
Comment (tbosch) [inline modules/angular2/src/core/application.js]: Move both in one line

File: modules/angular2/src/core/compiler/compiler.js [hunk 2]
```diff
@@ -149,7 +148,7 @@ export class Compiler {
   }
 
   // TODO(vicb): union type return ProtoView or Promise<ProtoView>
-  _compileTemplate(template: Template, tplElement: Element, component: Type) {
+  _compileTemplate(template: Template, tplElement, component: Type) {
```
Comment (tbosch) [inline modules/angular2/src/core/compiler/compiler.js]: Could we keep the type `Element`?

File: modules/angular2/src/dom/html5_adapter.dart [hunk 3]
```diff
@@ -0,0 +1,229 @@
+library angular2.dom.html5adapter;
```
Comment (tbosch) [inline modules/angular2/src/dom/html5_adapter.dart]: Could we rename this to `html5lib_adapter` / `Html5LibDomAdapter`?

File: modules/angular2/test/dom/dom_adapter.server.spec.dart [hunk 4]
```diff
@@ -0,0 +1,19 @@
+library angular2.dom.html5adapter.test;
```
Comment (tbosch) [inline modules/angular2/test/dom/dom_adapter.server.spec.dart]: should this file be called `Html5libDomAdapter.server.spec.dart`?

Comment (tbosch) [inline modules/angular2/test/dom/dom_adapter.server.spec.dart]: Should we add more tests here? If not, a comment to explain would be nice

File: modules/angular2/test/core/compiler/compiler_html5.server.spec.dart [hunk 5]
```diff
@@ -0,0 +1,9 @@
+library angular2.compiler.html5.test;
```
Comment (tbosch) [inline modules/angular2/test/core/compiler/compiler_html5.server.spec.dart]: Rename to `compiler_html5lib.server.spec.dart`

File: modules/angular2/src/facade/browser.dart [hunk 6]
```diff
@@ -0,0 +1,38 @@
+/**
```
Comment (tbosch) [inline modules/angular2/src/facade/browser.dart]: Could this be merged with the `dom_adapter` implementations?

Comment (tbosch) [inline modules/angular2/src/facade/browser.dart]: Only export `gc`, `window` and `document`. Import the types in the `dom_adapter` directly from the underlying libs.

File: modules/angular2/src/dom/html5_adapter.dart (hunk 3 above)
Comment (yjbanov) [inline modules/angular2/src/dom/html5_adapter.dart]: Done

File: modules/angular2/src/facade/browser.dart (hunk 6 above)
Comment (yjbanov) [inline modules/angular2/src/facade/browser.dart]: re: merging into dom_adapter. No, these things are specific to browser environment.
re: exporting only what's needed. Done.

File: modules/angular2/test/core/compiler/compiler_html5.server.spec.dart (hunk 5 above)
Comment (yjbanov) [inline modules/angular2/test/core/compiler/compiler_html5.server.spec.dart]: Done

File: modules/angular2/test/dom/dom_adapter.server.spec.dart (hunk 4 above)
Comment (yjbanov) [inline modules/angular2/test/dom/dom_adapter.server.spec.dart]: Done.
Done.

File: modules/angular2/src/core/application.js (hunk 1 above)
Comment (yjbanov) [inline modules/angular2/src/core/application.js]: These are different modules.

**Pull request closed without merging**